clvm-derive/src/from_clvm.rs

```diff
@@ -0,0 +1,51 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{parse_quote, Data, DataStruct, DeriveInput, Fields};
+
+use crate::helpers::{add_trait_bounds, crate_ident, parse_args, Repr};
+
+pub fn from_clvm(mut ast: DeriveInput) -> TokenStream {
+    let args = parse_args(&ast.attrs);
+    let crate_name = crate_ident();
+
+    let fields = match &ast.data {
+        Data::Struct(DataStruct {
+            fields: Fields::Named(fields),
+            ..
+        }) => &fields.named,
+        _ => panic!("expected a struct with named fields"),
```

Comment:
there's no reason this wouldn't work for tuples though, right?

Reply:
We can support tuple structs, numeric enums, data enums, and maybe option-like enums. The complexity scales the more we add support for, so I haven't implemented those yet. I want to make sure we like this before adding more functionality if possible